### DISCUSSION TITLE
Power problem with my build

### DISCUSSION THREAD

**Discussion by Ourbudd** [Resinchem/ESP-Parking-Assistant#27] [Troubleshooting and Assistance]
1st off Thanks for this build lesson. This is my 1st Arduino project and it has been a nice project with great instructions.

I have built the unit just like you said with the parts you suggested. I am having one problem and haven't figured out what is causing it. When I boot it up on the power supply it generally does not do anything, I checked voltages on both sides and they right in line. After testing numerous times it came on and worked fine but after disconnecting power and trying again I couldn't get it to work again. The light blinks but occasionally and random amounts from what I can tell. I looked the board over with a scope and don't see any issues but not sure what is causing the problem. Then when I was testing trying to figure it out I disconnected the power supply and connected it to a battery pack with a micro USB and it came on and worked, every time I tried it? I then tried back and forth between the 2 different power sources and one more time it worked on the power supply after about a minute or so but most of the time it doesn't. I checked voltages on both rails on both power supplies and they are very similar 3.3 and 5, I also took voltage readings on the logic level shifter and most seemed good but on LU4 when on the battery source I get 0.25 to 0.65 V and when on the wall power supply it gives me about 3.3 V, I tried many times and it is basically the same. When the unit did work on the wall power supply the voltage was low like on the battery. I tried moving things around and looking deeper with the digital scope but can't fine the problem. Any help is appreciated and being new I am sure some of this could have been said better. I will attach pictures of the board connections. Thanks,   Tim
![20240206_202206](https://github.com/Resinchem/ESP-Parking-Assistant/assets/159212715/2f7dc20b-df23-469d-b33d-10067c911b63)
![20240206_202146](https://github.com/Resinchem/ESP-Parking-Assistant/assets/159212715/ddc86d3d-e258-4ecc-aaed-1be9b6802e4d)

**Comment by Resinchem** [comment]
I've converted this over to a discussion (for now at least) as it doesn't appear to be an issue with the firmware itself.... especially since you said the system works via USB.  If we do determine that it is a firmware issue, I will gladly move it back to an open issue.  I'm trying to reserve the issue for problems related to the firmware (things I need to fix).  By initially posting this in discussions, it will remain available for others to see if they are experiencing a similar issue. 

All that being said, from what you describe this does appear to be a power or wiring issue.  As far as I can tell from the photos, the wiring looks OK.  Can I ask which particular brand/vendor that you are using for the logic level shifter?  I only ask because I've had a number of folks report back to me that there has been some sort of change with the shifters supplied by HiLetgo.  I've tried to update all my links for the shifter to a different vendor, but wanted to check whether you may have gotten one of the 'new', yet problematic, shifters from HiLetgo.

However, the fact that it works via USB says that it isn't necessarily a shifter issue.. but something with the board wiring.. since the shifter would be the same in both cases.  Were you able to test this via a breadboard before creating the soldered version?  Or do you have an extra D1 Mini and shifter where you could try duplicating the system, but on a breadboard?  I'm just trying to eliminate the possibility that it could be either a faulty component (D1 Mini or level shifter) or a soldered wiring issue.  Short tests are fine, but you generally do not want to power the system via USB on the D1 Mini as even something like 20-30 LEDs could try to pull enough current to damage or destroy the board when powering the LEDs via the 5V pin on the ESP board.

I'm very happy to try to help you get this resolved.  But it may just be a bit of process of elimination to narrow down where the problem is occurring.  And just for my own info, can you let me know which version/release of the firmware you installed as I just released a new version yesterday... but again, I don't think this is firmware related.

**Reply by Ourbudd** [reply]
Here, VKLSVAN 10PCS 4 Channels IIC I2C Logic Level Converter are the shifters, the D1 is ACEIRMC 5pcs ESP8266 ESP-12 ESP-12F NodeMcu Mini D1, both off of your links. I did see your new download the day after I got this one, I thought I would flash with that and see if any different if not I was going to changeout the D1 Mini with a different one and try that next. I didn't try on a breadboard before so not sure on that but I can. I will let you know what I attempt soon.  Thanks,  Tim

**Reply by Resinchem** [reply]
Yeah... I doubt it is the shifter or the D1 Mini... but you never know.  The only reason I recommended a breadboard version is that it would eliminate any potential issues with bad/bridged solder joints, etc.  If it is working with USB, but not via a direct 5V connection (or intermittently), then that does point to something with the soldering/wiring.  

I know it is water under the bridge at this point, but I always like to build a breadboard version of anything I'm eventually going to solder first... even for my WLED controllers that I have built a dozen times.  That way I can actually test not only my wiring logic, but the components themselves.  It is entirely possible to get a bad ESP board or other component from time to time, and it really sucks to find out you've got a bad component _after_ you've built the controller and soldered everything together!  So, I'll breadboard all the components I'm going to actually use and make sure they are all working... then I'll solder those same verified components into a permanent version.

The latest firmware just adds Home Assistant MQTT discovery and fixes a couple of small bugs/issues that were outstanding related to length of MQTT fields and the wifi AP/hotspot continuing to broadcast after connecting to local wifi.  Probably nothing that would impact the issue you are having.  But it's probably not a bad idea to use the most recent firmware anyway.

Please do let me know what you find out.

**Reply by Ourbudd** [reply]
I was just going to install and power from the USB but just couldn't do it. I went back and built it again on a breadboard as suggested and it worked fine so I decided to just rebuild the whole board and that also works fine. I still don't see any issue on the other but feel better with it working properly. I have it installed and it's working great. Only problem I have had is sometimes I can't connect on the computer? For some reason I lean towards a poor connection on the TFMini but cant say for sure, it works good just sometimes I can't connect? For a first project as I said this was easy on my part as your project was laid out so nicely and explained to the tee including the 3D print box. Thanks for it and look forward to something new soon. 
  I do intend to start doing things on the our Google Home (Home Assist) but that will be another time soon.
One side question is what wire do you use for your stranded, I had some 18 awg stranded but most of \what I see out there they are using say 22 awg but 60 strands? Just curious.
![20240210_190238](https://github.com/Resinchem/ESP-Parking-Assistant/assets/159212715/18a565d1-d843-4ef2-a725-91412d6b48a3)
![20240210_190223](https://github.com/Resinchem/ESP-Parking-Assistant/assets/159212715/c91e6db0-591f-4bb0-a51d-4c4ac1edee3e)

**Reply by Resinchem** [reply]
Congrats on finishing the build!  It's sometimes hard to determine why a particular device or build doesn't work quite right.  It could be an intermittent connection, sketchy solder joint, faulty component or any of many other reasons.  But as I mentioned, I do always at least build a breadboard version so I can verify the components before I ever create a soldered version.

As far as not being able to connect to the web app at times... again, could be multiple reasons, but assuming this is in your garage, maybe your wifi signal isn't real strong out there?  It could be that the D1 Mini is struggling to maintain a connection.  The system will continue to work if wifi drops, but you wouldn't be able to connect to the web interface.  Do note that the web site can sometimes take a little while to respond.  The little D1 Mini isn't all that powerful when it comes to being a web server... and it is running other tasks as well.  So if you don't get an immediate response from the web server, give it a few moments.  Sometimes mine responds right away... at other times, it might take 10-15 seconds to respond.  And that is with very strong WiFi (I have an access point mounted right in the garage).  A weak signal could contribute to a slower response as well.  But it could be something else as well!  Again, if you aren't yet using MQTT/Home Assistant, then the wifi drops are not a big issue for now.

As far as wire gauge,  it really depends.  I do try to use no smaller than 18 gauge for anything that is going to carry power.  But for things like push buttons, sensors, etc.  I might use something like 22 or even 24 gauge.  The length of the wire run is also a determining factor.  The longer the run, the larger wire... as a general rule.  But as you likely have found out, even 18 gauge can be a bit tricky to get through the small holes of an ElectroCookie board with creating stray strands that could lead to a short or other issues.  Often, if I'm creating a connector so that I can easily disconnect the controller, I might use a very short of run of 20 gauge wired between the ElectroCoookie and whatever connector I'm using (barrel, spade, JST, etc.) but then use the larger 18 gauge from the other connector to the power supply.

But I keep everything from 10 gauge all the way down to 26 gauge on hand... stranded and solid.  What I use really depends on that particular connection/component and the amount of voltage/current in play.  I'd say for ElectroCookie connections, 20 and 22 gauge are my most common wire sizes.

![20240212_072622](https://github.com/Resinchem/ESP-Parking-Assistant/assets/55962781/607a6bec-9a96-4bff-902c-059244245478)

Again congrats.  I'm working on some additional improvements on the parking assistant... primarily related to the wifi onboarding process and being able to accept special characters in the WiFi SSID and password.  Not sure when it will be released.  Until then, don't hesitate to reach back out if you have any other questions or problems.
